src/main.rs

```diff
@@ -83,36 +83,133 @@ fn read_input(input_path: &path::PathBuf) -> Result<SetOntology<RcStr>, Box<dyn
+                                    );
+                                }
+                                _ => {}
+                            },
+                            _ => {}
+                        });
+
+                        let invalid_whelk_axioms = crate::translate_ontology(&io);
+
+                        invalid_whelk_axioms.iter().map(|a| Rc::deref(a)).for_each(|a| match a {
+                            wm::Axiom::ConceptInclusion(ci) => match (Rc::deref(&ci.subclass), Rc::deref(&ci.superclass)) {
+                                (Concept::AtomicConcept(sub), Concept::AtomicConcept(sup)) => {
+                                    let subclass_deref = ci.subclass.deref();
+                                    let supclass_deref = ci.superclass.deref();
+                                    assert!(!whelk_subs_by_subclass.contains_key(subclass_deref));
+                                    assert!(!whelk_subs_by_subclass.contains_key(supclass_deref));
+                                }
+                                _ => {}
+                            },
+                            _ => {}
+                        });
+                    }
+                    (Some(ao), Some(eo), None) => {
+                        let asserted_whelk_axioms = crate::translate_ontology(&ao);
+
+                        let whelk = whelk::whelk::reasoner::assert(&asserted_whelk_axioms);
+                        let whelk_subs_by_subclass = whelk.closure_subs_by_subclass;
+                        // whelk_subs_by_subclass.iter().for_each(|a| println!("subclass: {:?}", a));
+
+                        let entailed_whelk_axioms = crate::translate_ontology(&eo);
+                        entailed_whelk_axioms.iter().map(|a| Rc::deref(a)).for_each(|a| match a {
+                            wm::Axiom::ConceptInclusion(ci) => match (Rc::deref(&ci.subclass), Rc::deref(&ci.superclass)) {
+                                (Concept::AtomicConcept(sub), Concept::AtomicConcept(sup)) => {
+                                    let subclass_deref = ci.subclass.deref();
+                                    let supclass_deref = ci.superclass.deref();
+                                    assert!(whelk_subs_by_subclass.contains_key(subclass_deref));
+                                    assert!(whelk_subs_by_subclass.contains_key(supclass_deref));
+                                    let values_by_subclass = whelk_subs_by_subclass.get(subclass_deref);
+                                    assert!(values_by_subclass.is_some(), "{}", format!("sub class key is not found: {:?}", subclass_deref));
+                                    assert!(
+                                        values_by_subclass.unwrap().contains(supclass_deref),
+                                        "{}",
+                                        format!("super class is not contained in sub class set of values: {:?}:{:?}", subclass_deref, supclass_deref)
```

Comment:
This is duplicating code from above. I would like to just have it in one place. Both the entailed and invalid files should be optional. Always run the reasoner on the asserted file. If the entailed file is present, test it against the inferences. If the invalid file is present, test it against the inferences.